### PR TITLE
Streaming response and Petstore file server

### PR DESCRIPTION
**Pull request by samiyuru**
Note: Please do not merge this until the petstore-fileserver is tested in the kubernetes cluster

### CODE REVIEW

**Comment by afkham** [comment]
Did you complete testing this with a Kubernetes?

**Comment by lakwarus** [comment]
Still I'm getting build failure. Seems you have accidentally added CoreOS files. Please remove those. see following error

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.5.1:attached (distribution) on project wso2mss: Failed to create assembly: Error creating assembly archive mss: IOException when zipping rwso2mss-1.0.0-SNAPSHOT/samples/petstore/deployment/VirtualBox VMs/kubernetes-vagrant-coreos-cluster_node-02_1448344726601_36767/coreos_production_vagrant_image.vmdk: wso2mss-1.0.0-SNAPSHOT/samples/petstore/deployment/VirtualBox VMs/kubernetes-vagrant-coreos-cluster_node-02_1448344726601_36767/coreos_production_vagrant_image.vmdk's size exceeds the limit of 4GByte. -> [Help 1]

**Comment by afkham** [comment]
Build is working fine for me now. There are no coreos files
https://github.com/wso2/product-mss/tree/master/samples/petstore/deployment/kubernetes-vagrant-coreos-cluster

On Thu, Nov 26, 2015 at 1:17 PM, lakwarus <notifications@github.com> wrote:

> Still I'm getting build failure. Seems you have accidentally added CoreOS
> files. Please remove those. see following error
>
> [ERROR] Failed to execute goal
> org.apache.maven.plugins:maven-assembly-plugin:2.5.1:attached
> (distribution) on project wso2mss: Failed to create assembly: Error
> creating assembly archive mss: IOException when zipping
> rwso2mss-1.0.0-SNAPSHOT/samples/petstore/deployment/VirtualBox
> VMs/kubernetes-vagrant-coreos-cluster_node-02_1448344726601_36767/coreos_production_vagrant_image.vmdk:
> wso2mss-1.0.0-SNAPSHOT/samples/petstore/deployment/VirtualBox
> VMs/kubernetes-vagrant-coreos-cluster_node-02_1448344726601_36767/coreos_production_vagrant_image.vmdk's
> size exceeds the limit of 4GByte. -> [Help 1]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wso2/product-mss/pull/63#issuecomment-159838994>.
>



-- 
*Afkham Azeez*
Director of Architecture; WSO2, Inc.; http://wso2.com
Member; Apache Software Foundation; http://www.apache.org/
* <http://www.apache.org/>*
*email: **azeez@wso2.com* <azeez@wso2.com>
* cell: +94 77 3320919blog: **http://blog.afkham.org*
<http://blog.afkham.org>
*twitter: **http://twitter.com/afkham_azeez*
<http://twitter.com/afkham_azeez>
*linked-in: **http://lk.linkedin.com/in/afkhamazeez
<http://lk.linkedin.com/in/afkhamazeez>*

*Lean . Enterprise . Middleware*

File: carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/mime/MimeMapper.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            mimeMap.load(inputStream);
+            try {
+                inputStream.close();
+            } catch (IOException e) {
+                log.warn("Could not close input stream", e);
+            }
+        }
+    }
+
+    public static String getMimeType(String extension) throws MimeMappingException {
```
Comment (pamod) [inline carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/mime/MimeMapper.java]: Need to include method level comment  ?

Comment (afkham) [inline carbon-mss/components/org.wso2.carbon.mss/src/main/java/org/wso2/carbon/mss/internal/mime/MimeMapper.java]: Why? Isn't it obvious from the method name?

**Pull request merged**